dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/Store.java

```diff
@@ -42,6 +42,7 @@ public Expression getMemValue() {
 
     public void setMemValue(Expression value) {
         this.value = value;
+        setAccessType(value.getType());
```

Comment:
This is dangerous, because changing the mem value via `transformExpressions` can still give you mismatches between the expression type and the access type.
If we want them to always align, just overwrite `getAccessType()` and remove the concrete accessType member from the abstract base class.
We can do the same for loads.

Reply:
Now that you mention it, we may not want them to always align.  `Tearing` can adjust the access type explicitly.